wireless_modules/sensors/gps_sensor.py

```diff
@@ -0,0 +1,63 @@
+from machine import UART
+import uasyncio as asyncio
+
+from micropyGPS import MicropyGPS
+
+from sensor_base import Sensor
+
+
+class GpsSensor(Sensor):
+    """
+    A GPS sensor implementing NMEA-0183 connected via UART, such as the u-blox NEO 6M.
+    """
+
+    def __init__(self, uart_channel: int):
+        """
+        Begin reading data from the GPS over UART.
+        :param uart_channel: The UART channel that the GPS is connected to.
```

Comment:
Is this different to a pin on the ESP32?

Reply:
Yep! The ESP32 has 3 UART controllers, each is hardwired to a pair of pins. More about UART [here](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/uart.html), or you can see 2 of the UART channels available on this pin diagram below. (not sure where UART1 is... I'm wondering if it's connected to USB on the devboard or something like that)
![image](https://user-images.githubusercontent.com/17876556/105644381-60d69a80-5ee9-11eb-8ae6-700f0b8cc09f.png)